python/marvin/api/rss.py

```diff
@@ -109,32 +135,52 @@ def get(self, args, name):
         # Pop any args we don't want going into Rss
         args = self._pop_args(args, arglist='name')
 
-        rss, results = _getRSS(name, **args)
-        self.update_results(results)
+        rss, res = _getRSS(name, **args)
+        self.update_results(res)
 
         if rss:
-            # For now we don't return anything here, maybe later.
-            self.results['data'] = {}
+
+            try:
+                nsa_data = rss.nsa
+            except (MarvinError, BrainError):
+                nsa_data = None
+
+            wavelength = (rss._wavelength.tolist() if isinstance(rss._wavelength, numpy.ndarray)
+                          else rss._wavelength)
+
+            obsinfo = io.StringIO()
+            rss.obsinfo.write(format='ascii', filename=obsinfo)
+            obsinfo.seek(0)
+
+            self.results['data'] = {'plateifu': name,
+                                    'mangaid': rss.mangaid,
+                                    'ra': rss.ra,
+                                    'dec': rss.dec,
+                                    'header': rss.header.tostring(),
+                                    'redshift': nsa_data.z if nsa_data else -9999,
+                                    'wavelength': wavelength,
+                                    'wcs_header': rss.wcs.to_header_string(),
+                                    'nfibers': rss._nfibers,
+                                    'obsinfo': obsinfo.read()}
 
         return jsonify(self.results)
 
-    @route('/<name>/fibers/', methods=['GET', 'POST'], endpoint='getRSSAllFibers')
+    @route('/<name>/fibers/<fiberid>', methods=['GET', 'POST'], endpoint='getRSSFiber')
     @av.check_args()
-    def getAllFibers(self, args, name):
-        """Returns a list of all the flux, ivar, mask, and wavelength arrays for all fibres.
+    def getFiber(self, args, name, fiberid):
```

Comment:
What was the rationale behind getting rid of `getAllFibers`?   If I go to `/<name>/fibers/` without specifying a fiberid, will I get all the fibers?

Reply:
No, we could reimplement it. In general I'm a bit wary about API calls that return 50 MB responses but maybe that's ok ... It's not very complicated to add but also I don't have a use for it in the way I implemented the Tool.